### PR TITLE
1. Accidentally deleted a task from the repository.

### PR DESCRIPTION
**Pull request by Sasha-Vradiy**
2. Found an error in the test.
3. Removed unnecessary dependency.
4. Caught the error exactly where it occurred.
5. Returned String where it was needed.
6. Returned the read information.
7. Renamed the method.

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,76 @@
...
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileData = readFromFile(fromFileName);
+        String report = createReport(fileData);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String readFile) {
+        StringBuilder builder = new StringBuilder();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(readFile));
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:26]: **try-with-resources**
Remember, if you are using classes that implement an AutoCloseable interface, we should use it with try-with-resources.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,75 @@
...
+            String value = reader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't read data from file", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data to file", e);
+        }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:35]:
```suggestion
        } catch (Exception e) {
            throw new RuntimeException("Can't read data from file: " + fileName, e);
        } 
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,75 @@
...
+                } else {
+                    buyCount += Integer.parseInt(listFromFile[COUNT_BY_INDEX]);
+                }
+            }
+        }
+        int result = supplyCount - buyCount;
+        return report(supplyCount, buyCount, result);
+    }
+
+    private String report(int supplyCount, int buyCount, int result) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:57]: delete this method - do it inside `createReport`
**report** is a bad name for the method. Method name should have verb.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+    private static final int SEARCH_BY_INDEX = 0;
+    private static final int COUNT_BY_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileData = readFromFile(fromFileName);
+        String report = createReport(fileData);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String readFile) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:23]:
```suggestion
    private String readFromFile(String fileName) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,75 @@
...
+                + "," + supplyCount
+                + System.lineSeparator()
+                + BUY + ","
+                + buyCount
+                + System.lineSeparator()
+                + RESULT + ","
+                + result;
+    }
 
+    private void writeToFile(String toFileName, String report) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:68]: This method can write any text, not only report. Remember Abstraction.
```suggestion
    private void writeToFile(String fileName, String text) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,75 @@
...
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName, true))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file", e);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:72]: make msg informative
image

File: src/test/java/core/basesyntax/WorkWithFileTest.java [hunk 7]
```diff
@@ -73,8 +73,6 @@ public void getStatisticAboutBanana() {
     @Test
     public void getStatisticAboutBananaRepeatedMethodCall() {
         workWithFile.getStatistic("banana.csv", BANANA_RESULT_FILE);
-        workWithFile.getStatistic("banana.csv", BANANA_RESULT_FILE);
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/WorkWithFileTest.java:76]: revert test changes)
image

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,7 +1,75 @@
...
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int SEARCH_BY_INDEX = 0;
+    private static final int COUNT_BY_INDEX = 1;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]: operation_index, quantity_index

**Review by Rommelua — CHANGES_REQUESTED** [review]
Pay more attantion on naming 
image

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 9]
```diff
@@ -1,7 +1,75 @@
...
+        return builder.toString();
+    }
+
+    private String createReport(String readString) {
+        int supplyCount = 0;
+        int buyCount = 0;
+        String[] informationRead = readString.split(System.lineSeparator());
+        for (String infoFromFile : informationRead) {
+            String[] listFromFile = infoFromFile.split(",");
+            for (int i = 1; i < listFromFile.length; i++) {
```
[the remark this answers is not in the document]
Comment (Sasha-Vradiy) [reply src/main/java/core/basesyntax/WorkWithFile.java:45]: Попередній ментор сказав що потрібен.

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 3 above)
Comment (Sasha-Vradiy) [reply src/main/java/core/basesyntax/WorkWithFile.java:57]: Назву методу вказав попередній ментор.

File: src/test/java/core/basesyntax/WorkWithFileTest.java (hunk 7 above)
Comment (Sasha-Vradiy) [reply src/test/java/core/basesyntax/WorkWithFileTest.java:76]: Тоді скажіть як записати інформацію в файл знову, якщо він не пустий вже, воно ж дублуюється.

**Pull request closed without merging**